### PR TITLE
refactor(core): add conway core types

### PR DESCRIPTION
**Pull request by mchappell**
# Context

Core types need to be updated for the SDK to support Voltaire governance

# Proposed Solution

- Extract core types from [conway cddl](https://github.com/input-output-hk/cardano-ledger/blob/master/eras/conway/test-suite/cddl-files/conway.cddl)

- Add the necessary serialization classes to support Conway era.

# Important Changes Introduced

- Added additional core types to match conway era cddl
- New serialization classes conway era were added

### CODE REVIEW

File: packages/core/src/Cardano/types/Certificate.ts [hunk 1]
```diff
@@ -10,10 +11,102 @@ export enum CertificateType {
...
+export interface NewStakeAddressCertificate {
+  __typename: CertificateType.Registration | CertificateType.Unregistration;
+  stakeKeyHash: Crypto.Ed25519KeyHashHex;
+  deposit: Lovelace;
+}
+
+// Delegation
+export interface VoteDelegationCertificate {
+  __typename: CertificateType.VoteDelegation;
+  drep: DelegateRepresentative;
```
Comment (mirceahasegan) [inline packages/core/src/Cardano/types/Certificate.ts:42]: Missing stake credential field `vote_deleg_cert = (9, stake_credential, drep)`
```diff
+ stakeKeyHash: Crypto.Ed25519KeyHashHex;
```

File: packages/core/src/Cardano/types/Generic.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+export interface Fraction {
```
Comment (rhyslbw) [inline packages/core/src/Cardano/types/Generic.ts:1]: `Generic` is an overloaded term in TypeScript when referring to types, so probably best to rename this module

File: packages/core/src/Cardano/types/Governance.ts [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+  vote: Vote;
+  anchor: Anchor | null;
+};
+
+export type VotePayload = {
+  governanceActionId: GovernanceAction;
+  votingProcedure: VotingProcedure;
+};
+
+export type Voter = Credential;
```
Comment (mirceahasegan) [inline packages/core/src/Cardano/types/Governance.ts:118]: `Voter` could be a keyhash or a script hash, but they could represent different things (e.g. CC keyhash, DRep keyhash or stakingPool keyhash). We'll need to provide a credential type that holds this information.
```
; Constitutional Committee Hot KeyHash: 0
; Constitutional Committee Hot ScriptHash: 1
; DRep KeyHash: 2
; DRep ScriptHash: 3
; StakingPool KeyHash: 4
voter =
  [ 0, addr_keyhash
  // 1, scripthash
  // 2, addr_keyhash
  // 3, scripthash
  // 4, addr_keyhash
  ]
```

File: packages/core/src/Cardano/types/Governance.ts [hunk 4]
```diff
@@ -110,17 +110,10 @@ export type VotingProcedure = {
   anchor: Anchor | null;
 };
 
-export type VotePayload = {
-  governanceActionId: GovernanceAction;
-  votingProcedure: VotingProcedure;
-};
-
-export type Voter = Credential;
+// Do we need to know what type of credential the Voter is?
```
Comment (mirceahasegan) [inline packages/core/src/Cardano/types/Governance.ts:113]: I vote "yes". Depending on the type of credential, we determine if we can sign it and what is the derivation path.
```
; Constitutional Committee Hot KeyHash: 0
; Constitutional Committee Hot ScriptHash: 1
; DRep KeyHash: 2
; DRep ScriptHash: 3
; StakingPool KeyHash: 4
voter =
  [ 0, addr_keyhash
  // 1, scripthash
  // 2, addr_keyhash
  // 3, scripthash
  // 4, addr_keyhash
  ]
```

File: packages/core/src/Cardano/types/Certificate.ts [hunk 5]
```diff
@@ -10,10 +11,103 @@ export enum CertificateType {
...
+  VoteDelegation = 'VoteDelegationCertificate',
+  StakeVoteDelegation = 'StakeVoteDelegationCertificate',
+  StakeRegistrationDelegation = 'StakeRegistrationDelegateCertificate', // delegate or delegation??
+  VoteRegistrationDelegation = 'VoteRegistrationDelegateCertificate', // same as above
+  StakeVoteRegistrationDelegation = 'StakeVoteRegistrationDelegateCertificate',
+  AuthorizeCommitteeHot = 'AuthorizeCommitteeHotCertificate',
+  ResignCommitteeCold = 'ResignCommitteeColdCertificate',
+  RegisterDelegateRepresentative = 'RegisterDelegateRepresentativeCertificate',
+  UnregisterDelegateRepresentative = 'UpdateDelegateRepresentativeCertificate',
+  UpdateDelegateRepresentative = 'UpdateDelegateRepresentativeCertificate'
```
Comment (mirceahasegan) [inline packages/core/src/Cardano/types/Certificate.ts:29]: Typo:
```diff
-  UnregisterDelegateRepresentative = 'UpdateDelegateRepresentativeCertificate',
+  UnregisterDelegateRepresentative = 'UnregisterDelegateRepresentative',
UpdateDelegateRepresentative = 'UpdateDelegateRepresentativeCertificate'
```

**Review by mirceahasegan — APPROVED** [review]

File: packages/core/src/Cardano/types/Generic.ts (hunk 2 above)
Comment (mchappell) [reply packages/core/src/Cardano/types/Generic.ts]: Fixed in `08bf2b3ab0171306eeec830c97d025c9f6e1e057` called `UtilityTypes` unless there is a more descriptive name preferred

**Review by rhyslbw — APPROVED** [review]
Ready for squashing

**Comment by rhyslbw** [comment]
Actually, there's [failing unit tests](https://github.com/input-output-hk/cardano-js-sdk/actions/runs/6307892243/job/17125222003?pr=911)

**Review by rhyslbw — APPROVED** [review]

**Pull request merged**